framework/BaseClasses/PluginReadyEntity.py

```diff
@@ -77,3 +81,24 @@ def parseXML(self, xml):
     paramInput = self.getInputSpecification(xml=xml)()
     paramInput.parseNode(xml)
     return paramInput
+
+  @abstractmethod
+  def _getInterface(self):
+    """
+      Return the interface associated with this entity.
+      @ In, None
+      @ Out, _getInterface, object, interface object
+    """
+
+  ########
+  #
+  # Utilities
+  #
+  def isInstanceString(self, toCheck):
+    """
+      Use string type names to check if associated metric is one of those whose names are provided.
+      @ In, toCheck, list, list of names of viable types
+      @ Out, isInstanceString, bool, True if interface name matches one of the provided options
+    """
+    viable = tuple([self.interfaceFactory.returnClass(check) for check in np.atleast_1d(toCheck)])
```

Comment:
I am not aware of an "atleast_1d" option outside numpy; if I run that line above and the developer sends in a single string, then it will parse the string letter by letter. The array assures that does not happen.